src/app/pages/record/add-event/add-event.component.ts

```diff
@@ -12,14 +14,12 @@ import { RecordService } from '../record.service';
 })
 export class AddEventComponent implements OnInit, OnDestroy {
 
-
   public form!: FormGroup;
-
   private sub: Subscription = new Subscription();
 
   constructor(
     @Inject(MAT_DIALOG_DATA) public data: ICategories[],
-    public dialogRef: MatDialogRef<any>,
+    public dialogRef: MatDialogRef<RecordComponent, HistoryComponent>,
```

Comment:
Dialog Overview example
https://material.angular.io/components/dialog/examples

Reply:
I know that there must be only one component in <>, but I'm using this modal twice, in two different components. Which one I should specify?

Reply:
Look very carefully at the specification, especially the example that I pointed out